src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let clonedState = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        const nextState = Object.assign({}, clonedState, action.extraData);
+
+        result.push(nextState);
```

Comment:
move it after `switch .. case` to not repeat

Reply:
Not resolved. Just move it after switch case construction to avoid code duplication. If you have any questions - feel free to ask them in `fe_chat`